File: backend/chat-store-service/src/main/java/com/example/chatstoreservice/message/ChatModel.java
package com.example.chatstoreservice.message;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import java.util.Date;

@Data
public class ChatModel {

    @JsonProperty("chat_id")
    private Long chatId;

    @JsonProperty("chat_type")
    private int chatType;

    @JsonProperty("room_id")
    private String roomId;

    @JsonProperty("sender")
    private String sender;

    @JsonProperty("message")
    private String message;

    @JsonProperty("chat_time")
    private String chatTime;

    @JsonProperty("chat_room_id")
    private Long chatRoomId;

    @JsonProperty("created_date")
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss", iso = DateTimeFormat.ISO.DATE_TIME)
//    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss.SSS", timezone = "Asia/Seoul")
    private Date createdDate;

    @JsonProperty("updated_date")
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss", iso = DateTimeFormat.ISO.DATE_TIME)
//    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Seoul")
    private Date updatedDate;
}

File: backend/chat-store-service/src/main/kotlin/com/example/chatstoreservice/message/ChatModel.kt
package com.example.chatstoreservice.message

import com.fasterxml.jackson.annotation.JsonProperty
import java.util.Date

/** Debezium 의 행 스냅샷(snake_case). 필드에 붙인 @JsonProperty 로 Kotlin 모듈 없는 ObjectMapper 도 읽는다. */
class ChatModel {
    @field:JsonProperty("chat_id")
    var chatId: Long? = null

    @field:JsonProperty("chat_type")
    var chatType: Int = 0

    @field:JsonProperty("room_id")
    var roomId: String? = null

    @field:JsonProperty("sender")
    var sender: String? = null

    @field:JsonProperty("message")
    var message: String? = null

    @field:JsonProperty("chat_time")
    var chatTime: String? = null

    @field:JsonProperty("chat_room_id")
    var chatRoomId: Long? = null

    @field:JsonProperty("created_date")
    var createdDate: Date? = null

    @field:JsonProperty("updated_date")
    var updatedDate: Date? = null

    override fun toString(): String =
        "ChatModel(chatId=$chatId, chatType=$chatType, roomId=$roomId, sender=$sender, message=$message, chatTime=$chatTime, chatRoomId=$chatRoomId, createdDate=$createdDate, updatedDate=$updatedDate)"
}
